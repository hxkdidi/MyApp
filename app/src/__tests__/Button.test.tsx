import React from 'react'
import Button from '../Button'
import { render, shallow, fireEvent } from 'react-native-testing-library'

let mockOpenURL = jest.fn()
jest.mock('Linking', () => ({
  openURL: mockOpenURL,
}))

describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot - Primary', () => {
      const { output } = shallow(<Button label="test" primary />)
      expect(output).toMatchSnapshot('Primary button snapshot')
    })

    it('should match to snapshot - Secondary', () => {
      const { output } = shallow(<Button label="test" primary={false} />)
      expect(output).toMatchSnapshot('Secondary button snapshot')
    })

    it('should render a Text', () => {
      const { getByTestId } = render(<Button label="test" primary={false} />)
      const text = getByTestId('text')
      expect(text.props.children).toEqual('test')
    })
  })

  describe('Event Handling', () => {
    const mockOnPress = jest.fn()
    beforeEach(() => {
      jest.clearAllMocks()
    })

    it('should call onPress', () => {
      const { getByTestId } = render(<Button label="test label" onPress={mockOnPress} />)
      fireEvent.press(getByTestId('touchable'))

      expect(mockOnPress).toHaveBeenCalled()
      expect(mockOnPress).toHaveBeenCalledTimes(1)
    })

    it('should call openURL if url is provided', () => {
      const { getByTestId } = render(<Button label="test label" url="https://www.test.com" />)
      fireEvent.press(getByTestId('touchable'))

      expect(mockOpenURL).toHaveBeenCalled()
      expect(mockOpenURL).toHaveBeenCalledTimes(1)
      expect(mockOpenURL).toHaveBeenCalledWith('https://www.test.com')
    })

    it('should not call openURL if url is nor provided', () => {
      const { getByTestId } = render(<Button label="test label" />)
      fireEvent.press(getByTestId('touchable'))

      expect(mockOpenURL).not.toHaveBeenCalled()
    })
  })
})
