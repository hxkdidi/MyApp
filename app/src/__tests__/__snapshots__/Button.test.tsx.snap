// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot - Primary: Primary button snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#38ba7d",
      "borderBottomColor": "#1e6343",
      "borderBottomWidth": 6,
      "borderWidth": 1,
      "height": 45,
      "justifyContent": "center",
      "marginLeft": 15,
      "marginRight": 15,
    }
  }
  testID="touchable"
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 17,
        "fontWeight": "600",
      }
    }
    testID="text"
  >
    test
  </Text>
</TouchableOpacity>
`;

exports[`Button Rendering should match to snapshot - Secondary: Secondary button snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#38ba7d",
        "borderBottomColor": "#1e6343",
        "borderBottomWidth": 6,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginLeft": 15,
        "marginRight": 15,
      },
      Object {
        "backgroundColor": "#f34541",
        "borderBottomColor": "#a43532",
      },
    ]
  }
  testID="touchable"
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 17,
        "fontWeight": "600",
      }
    }
    testID="text"
  >
    test
  </Text>
</TouchableOpacity>
`;
